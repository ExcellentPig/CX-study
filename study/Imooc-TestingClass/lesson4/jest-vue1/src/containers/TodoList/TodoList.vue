<template>
  <div>
    <Header @add="addUndoItem" />
 <!--   <ul>
    <li
    v-for="(item,index) in undoList"
    :key="index"
    >{{item}}
    </li>
    </ul> -->
    <UndoList
    :list="undoList"
    @delete="handleItemDelete"
    @status="changeStatus"
    @reset="resetStatus"
    @change="changeItemValue"
     />
  </div>
</template>

<script>
import axios from 'axios'
import Header from './components/Header.vue'
import UndoList from './components/UndoList.vue'

export default {
  name: 'TodoList',
  components: {
    Header,
    UndoList
  },
  props: {},
  data () {
    return {
      undoList: []
    }
  },
  mounted () {
    // axios.get('/getUndoList.json').then((res) => {
    //   this.undoList = res.data
    // }).catch((e) => {
    //   console.log(e)
    // })

    setTimeout(() => (
      axios.get('/getUndoList.json').then((res) => {
        this.undoList = res.data
      }).catch((e) => {
        console.log(e)
      })
    ), 5000)
  },
  methods: {
    addUndoItem (inputValue) {
      this.undoList.push({
        status: 'div',
        value: inputValue
      })
    },
    handleItemDelete (index) {
      this.undoList.splice(index, 1)
    },
    changeStatus (index) {
      const newList = []
      this.undoList.forEach((item, itemIndex) => {
        if (itemIndex === index) {
          newList.push({
            status: 'input',
            value: item.value
          })
        } else {
          newList.push({ status: 'div', value: item.value })
        }
        this.undoList = newList
      })
    },
    resetStatus () {
      const newList = []
      this.undoList.forEach((item, itenIndex) => {
        newList.push({ status: 'div', value: item.value })
      })
      this.undoList = newList
    },
    changeItemValue (obj) {
      this.undoList[obj.index].value = obj.value
    }
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="stylus">

</style>
