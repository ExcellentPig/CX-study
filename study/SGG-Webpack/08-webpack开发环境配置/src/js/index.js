// 引入 iconfont 样式文件
import '../css/iconfont.css';
import '../css/index.less'


function add (x, y) {
	return x + y
}

console.log(add(1, 2))