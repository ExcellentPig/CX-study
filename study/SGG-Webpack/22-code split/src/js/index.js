function sum (...args) {
	return args.reduce((p,c) => p+c, 0) // 0 是初始化值
}

/**
 * 通过js代码 将某文件打包成单独的chunk
 * import 动态导入语法： 能将某个文件单独打包
 * webpackChunkName: 'test' 设置文件名
 */
import (/* webpackChunkName: 'test' */'./test')
	.then((res) => {
		// 加载成功
		console.log(res)
	})
	.catch(() => {
		console.log('文件加载失败')
	})

console.log(sum(1,2,3,4,5,6))
