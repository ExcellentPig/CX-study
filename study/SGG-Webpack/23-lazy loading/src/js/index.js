console.log('index.js')

// import { mul } from './test.js'

// console.log(mul(3, 4))

document.getElementById('btn').onclick = function () {
	//console.log(mul(4, 4))
	// 懒加载  当文件需要使用时再加载
	// webpackPrefetch: true 预加载  会提前加载js文件  等其他资源加载完毕浏览器空闲了 再偷偷加载资源
	// 正常加载可以认为是并行加载(同一时间加载多个文件)
	import (/* webpackChunkName: 'test', webpackPrefetch: true */'./test').then(({mul}) => {
			console.log(mul(4, 5))
		})
}