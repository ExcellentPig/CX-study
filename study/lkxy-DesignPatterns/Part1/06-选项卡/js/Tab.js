function TabsFn () {
	this.lis = $('tab_header').getElementsByTagName('li')
	this.contents = $('tab_content').getElementsByClassName('dom');
}

TabsFn.prototype = {
	init: function () {
		this.setId()
		this.bindEvent()
	},
	setId: function () { //设置索引
		for (var i=0;i<this.lis.length;i++) {
			this.lis[i].id = i
		}
	},
	bindEvent: function () { // 绑定事件
		var self = this
		for (var i = 0; i < this.lis.length; i++) {
		    this.lis[i].onmouseover = function () {
		        // 2.1 所有的都不被选中
		        for(var j=0; j<self.lis.length; j++){
		            self.lis[j].className = '';
		            self.contents[j].style.display = 'none';
		        }
		        // 2.2 当前的li被选中
		        this.className = 'selected';
		        self.contents[this.id].style.display = 'block';
		    }
		}
	}
}