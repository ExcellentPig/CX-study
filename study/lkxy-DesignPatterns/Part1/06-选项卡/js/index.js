/*window.onload = function (ev) {
    // 1. 获取需要的标签
    var allLis = $('tab_header').getElementsByTagName('li');
    var allDom = $('tab_content').getElementsByClassName('dom');
    // console.log(allLis, allDom);

    // 2. 遍历监听
    for (var i = 0; i < allLis.length; i++) {
        var li = allLis[i];
        li.id = i;
        li.onmouseover = function () {
            // 2.1 所有的都不被选中
            for(var j=0; j<allLis.length; j++){
                allLis[j].className = '';
                allDom[j].style.display = 'none';
            }
            // 2.2 当前的li被选中
            this.className = 'selected';
            allDom[this.id].style.display = 'block';
        }
    }

};*/

var tab = new TabsFn();
tab.init();

/**
 * 根据id获取标签
 * @param {string}id
 * @returns {object}
 */
function $(id) {
    return typeof id === 'string' ? document.getElementById(id) : null;
}