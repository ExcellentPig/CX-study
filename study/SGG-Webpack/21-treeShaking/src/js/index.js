import { mul } from './test.js'
import '../css/index.css'

function sum (...args) {
	return args.reduce((p,c) => p+c, 0) // 0 是初始化值
}
console.log(mul(2, 4))

console.log(sum(1,2,3,4,5,6))
