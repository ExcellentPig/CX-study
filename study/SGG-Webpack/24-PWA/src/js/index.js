import { mul } from './test.js'
import '../css/index.css'

function sum (...args) {
	return args.reduce((p,c) => p+c, 0) // 0 是初始化值
}
console.log(mul(2, 4))

console.log(sum(1,2,3,4,5,6))

// 注册serviceworker
// 处理兼容问题
/*
"eslistConfig": {
    "extends": "airbnb-base",
	"env": {
		"browser": true
	}
  },
  
  eslint 不认识 window,navigator全局变量
	解决需要修改package.json配置 "browser": true
	
	sw代码必须运行在服务器上
		1、nodejs
		2、npm i server -g
		server -s build 启动服务器 将build目录下的所有资源暴露出去
*/
if ('serviceWorker' in navigator) {
	window.addEventListener('load', () => {
		navigator.serviceWorker.register('/service-worker.js')
			.then(() => {
				console.log('serviceworker注册成功')
			})
			.catch(() => {
				console.log('serviceworker注册失败')
			})
	})
}
