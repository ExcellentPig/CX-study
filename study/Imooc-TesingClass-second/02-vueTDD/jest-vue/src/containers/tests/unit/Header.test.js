import { shallowMount } from '@vue/test-utils'
import TodoList from '@/containers/TodoList/components/Header.vue'

import { findTestWrapper } from '../../../utils/testUtils.js'

it(`Header 包含 input 框`, () => {
  const wrapper = shallowMount(TodoList)
  const input = wrapper.find('[data-test="input"]')
  expect(input.exists()).toBe(true)
})

it(`Header input 框初始内容为空`, () => {
  const wrapper = shallowMount(TodoList)
  const inputValue = wrapper.vm.inputValue;
  expect(inputValue).toBe('')
})

it(`Header input数据发生变化，数据应该跟着变`, async () => {
  const wrapper = shallowMount(TodoList)
  const input = findTestWrapper(wrapper, 'input')
  await input.setValue('set khan value')
  const inputValue = wrapper.vm.inputValue;
  expect(inputValue).toBe('set khan value')
})

it(`Header input输入回车无内容时，无反应`, () => {
  const wrapper = shallowMount(TodoList)
  const input = findTestWrapper(wrapper, 'input')
  input.setValue('')
  input.trigger('keyup.enter')
  expect(wrapper.emitted().add).toBeFalsy()
})

it(`Header input输入回车有内容时，向外触发事件, 同时清空inputValue`, () => {
  const wrapper = shallowMount(TodoList)
  const input = findTestWrapper(wrapper, 'input')
  input.setValue('have content')
  input.trigger('keyup.enter')
  expect(wrapper.emitted().add).toBeTruthy()
  expect(wrapper.vm.inputValue).toBe('')
})

it(`Header 样式发生改变作提示`, () => {
  const wrapper = shallowMount(TodoList)
  expect(wrapper).toMatchSnapshot()
})
