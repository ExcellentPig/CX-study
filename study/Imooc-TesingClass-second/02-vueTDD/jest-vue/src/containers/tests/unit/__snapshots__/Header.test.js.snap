// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 样式发生改变作提示 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "__emits": Object {},
      "__props": Array [
        Object {},
        Array [],
      ],
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div
        class="header"
      >
        <div
          class="header-content"
        >
           TodoList 
          <input
            class="header-input"
            data-test="input"
            placeholder="请输入!"
          />
        </div>
      </div>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {},
      "config": Object {
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isCustomElement": [Function],
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "mixins": Array [
        Object {
          "__emits": null,
          "__props": Array [],
          "beforeCreate": [Function],
        },
        Object {
          "__emits": null,
          "__props": Array [],
          "beforeCreate": [Function],
        },
      ],
      "provides": Object {},
      "reload": [Function],
    },
    "_props": null,
    "_uid": 5,
    "component": [Function],
    "config": Object {
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isCustomElement": [Function],
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.0.2",
  },
  "__functionalEmits": Object {},
  "__setProps": [Function],
  "componentVM": Object {
    "__emitted": Object {},
    "addTodoItem": [Function],
    "inputValue": "",
  },
  "rootVM": Object {
    "__emitted": Object {},
  },
}
`;
